### PR TITLE
Fix scroll bar bottom position when dragging and fix high FPS rate when dragging scroll bar by mouse

### LINKED ISSUE #5840: Scroll bar shifts in the bottom edge while click
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

Scroll bar shifts a bit when we move it to a bottom edge of a short list and left click on it.

https://user-images.githubusercontent.com/55348946/190853391-833ad05c-2e76-4014-84d9-71bb27b27f66.mp4

### Save file

None.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by Districh-ru**
close #5840

This PR also makes the render of scrollbar only if scrollbar position was changed by mouse dragging to reduce FPS and CPU usage.

### CODE REVIEW

File: src/fheroes2/gui/ui_scrollbar.cpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ namespace fheroes2
         if ( _isVertical() ) {
             const int32_t scrollbarImageMiddle = height() / 2;
             const int32_t minYPos = _area.y + scrollbarImageMiddle;
-            const int32_t maxYPos = _area.y + roiHeight + height() - scrollbarImageMiddle - 1;
```
Comment (ihhub) [inline src/fheroes2/gui/ui_scrollbar.cpp:128]: Could you please verify that all scrollbars in the game behave properly after this change?

File: src/fheroes2/gui/interface_list.h [hunk 2]
```diff
@@ -468,6 +479,7 @@ namespace Interface
         int maxItems;
 
         fheroes2::Point ptRedraw;
+        fheroes2::Point _mousePosition;
```
Comment (ihhub) [inline src/fheroes2/gui/interface_list.h:482]: I suggest to name this member as **_prevMousePosition** to be even more clear about its purpose.

File: src/fheroes2/gui/interface_list.h [hunk 3]
```diff
@@ -387,12 +387,23 @@ namespace Interface
                 return true;
             }
             if ( le.MousePressLeft( _scrollbar.getArea() ) && ( _size() > maxItems ) ) {
+                const fheroes2::Point mousePosition = le.GetMouseCursor();
```
Comment (ihhub) [inline src/fheroes2/gui/interface_list.h:390]: What if we remember the position of the scrollbar before and then check where it should be after. If there are no change we don't draw. What do you think?

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I put few comments and a suggestion. Could you please take a look?

File: src/fheroes2/gui/ui_scrollbar.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/fheroes2/gui/ui_scrollbar.cpp:128]: Hi @ihhub, I've checked the scrollbars in Language/Video Mode selection, Load, Save, choose map, Kingdom view, Hero and Town scroll, Battle log, Town Portal dialogs. It works OK.
The problem was in rounding the result of division by 2. I've just simplified the expression: 
_area.y + roiHeight + height() - scrollbarImageMiddle - 1 = _area.y + roiHeight + height() - height() / 2 - 1 = _area.y + roiHeight + height() / 2 -1 = _area.y + roiHeight + scrollbarImageMiddle - 1.
And removed extra -1, which was added to patch this division rounding problem, which was leading to render the scrollbar one pixel down than its lowest position in some conditions.

File: src/fheroes2/gui/interface_list.h (hunk 2 above)
Comment (Districh-ru) [reply src/fheroes2/gui/interface_list.h]: As we don't check the mouse position now this part of code is removed.

**Review by ihhub — APPROVED** [review]

**Review by Branikolog — APPROVED** [review]
👍

**Comment by ihhub** [comment]
@Districh-ru , thanks a lot for this fix!

**Pull request merged**